### ISSUE TITLE
Check before use

### ISSUE THREAD

**Issue by mjmill** [FIRST-Team-2557-The-SOTABots/FRC_Robot2.0#11]
The variables start and end must be checked before they are used.

https://github.com/FIRST-Team-2557-The-SOTABots/FRC_Robot2.0/blob/3cc007571ec1ee8836c465c14ca1ee3d8da95e16/GetLatency/src/Detector.java#L49

**Comment by mjmill** [comment]
Preferably, near when the index is called.

Line 18 and 19 are good candidates.

**Comment by liloheinrich** [comment]
yes, you might get an index out of bounds error. end and start are on lines 18-19, so after that line, you could see if(LatencyString.substring(start+1, end+1) != null) before storing it.

**Comment by mjmill** [comment]
Something interesting here is by adding 1 it changes the value to 0.  That might not be valid all of the time, but it could be some of the time.

**Comment by liloheinrich** [comment]
what do you mean by adding 1 changes the value to 0?

**Comment by mjmill** [comment]
String::indexOf returns -1 when it can't find the item being searched for.  -1 + 1 = 0

**Comment by liloheinrich** [comment]
So could this work:
if(start > -1 && end + 1 < LatencyString.length()){
LatencyString.substring(start+1, end+1);
}

**Comment by mjmill** [comment]
Do you think end should be checked too?

**Comment by mjmill** [comment]
Oh, tricky.  Maybe make it clearer.  Easier to read is better.  Initially the check on end wasn't clear to me.

**Comment by liloheinrich** [comment]
this is more thorough and describes what .length() returns
int lastIndex = LatencyString.length() - 1; // 0 is counted as an index, ex. length 5 means last index is 4
if(start > -1 && end > -1 && start+1 <= lastIndex && end+1 <= lastIndex){
String substring = LatencyString.substring(start+1, end+1);
}
